graphql_base/types.py

```diff
@@ -27,6 +29,12 @@ def odoo_attr_resolver(attname, default_value, root, info, **args):
             return None
     elif isinstance(field, fields.Datetime):
         return fields.Datetime.context_timestamp(root, value)
+    elif (
+        not isinstance(field, fields._RelationalMulti)
+        and isinstance(value, models.BaseModel)
+        and not value
+    ):
```

Comment:
I do not immediately see in which case this part applies. Could you explain a little bit?

Reply:
It's to detect that it's a 2one relationship without a value. Those should return `None`.

If it's a 2many relationship (`isinstance(field, fields._RelationalMulti)`) then returning an empty recordset is fine, that'll get turned into an empty array like it should. (It might have been better to instead check in the schema if the field type is an array, but I couldn't figure out how to do that properly.)

And if it's not a recordset (`isinstance(value, models.BaseModel)`) then it's not a relation so the whole thing doesn't apply.